pkg/deploy/internal/setting/setting.go

```diff
@@ -34,12 +34,17 @@ import (
 	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/deploy/internal/extract"
 )
 
-func Deploy(ctx context.Context, settingsClient client.SettingsClient, properties parameter.Properties, renderedConfig string, c *config.Config, insertAfter string) (entities.ResolvedEntity, error) {
+func Deploy(ctx context.Context, settingsClient client.SettingsClient, properties parameter.Properties, renderedConfig string, c *config.Config) (entities.ResolvedEntity, error) {
 	t, ok := c.Type.(config.SettingsType)
 	if !ok {
 		return entities.ResolvedEntity{}, errors.NewConfigDeployErr(c, fmt.Sprintf("config was not of expected type %q, but %q", config.SettingsTypeID, c.Type.ID()))
 	}
 
+	var insertAfter string
+	if ia, ok := properties[config.InsertAfterParameter]; ok {
+		insertAfter = ia.(string)
+	}
+
```

Comment:
What about extracting this into a function in `pkg/deploy/internal/extract/configname.go`, _renaming that file_, and checking the type, like for `extract.Scope(...)` below?

Reply:
I am happy to create a function, but since it's not shared and only used in settings, I would not move it to a shared package. 
Both `extract.Scope` and `extract.ConfigName` are used in more than one config deploy.

Reply:
Additionally, the functionality will change again in the followup PR :)

Reply:
As we now have it in a function, I would be in favor of adding a type check for the cast to string.

Reply:
We discussed it offline, and we will add the type checking in the followup PR.